lms/djangoapps/certificates/views/webview.py

```diff
@@ -251,7 +251,9 @@ def _update_course_context(request, context, course, course_key, platform_name):
             platform_name=platform_name)
     # If language specific templates are enabled for the course, add course_run specific information to the context
     if CertificateGenerationCourseSetting.is_language_specific_templates_enabled_for_course(course_key):
-        fields = ['start', 'end', 'max_effort', 'content_language']
+        fields = ['content_language']
+        if CertificateGenerationCourseSetting.is_hours_of_effort_included_for_course(course_key):
```

Comment:
Sanity check: Does Django cache the queries made by all these static calls to `CertificateGenerationCourseSetting` methods? For example, on line 253, the call to `CertificateGenerationCourseSetting.is_language_specific_templates_enabled_for_course` makes a query for the latest `CertificateGenerationCourseSetting` entry for the course, then checks the `language_specific_templates_enabled` field. Does that query get cached and reused in the call on line 255, or does that call perform another query?

Reply:
I believe the second call performs another query.
Do you think it would be worthwhile to implement a method for the CertificateGenerationCourseSettings model that retrieves all of the settings?

Reply:
Yes, I think it would be worthwhile to do that. It should only take a minute and will prevent us from continuing the pattern of adding additional queries every time we need to check a new field on `CertificateGenerationCourseSetting`.